File: bitget-golang-sdk-api/pkg/model/mix/order/batchordersreq.go
package order

/**
 * @Author: bitget-sdk-team
 * @Date: 2022-09-30 10:46
 * @DES: place batch order request
 */
type BatchOrdersReq struct {
	/**
	 * Order data list
	 */
	OrderDataList []PlaceOrderBaseParam `json:"orderDataList"`
	/**
	 * Currency pair
	 */
	Symbol string `json:"symbol"`
	/**
	 * Deposit currency
	 */
	MarginCoin string `json:"marginCoin"`
}
type PlaceOrderBaseParam struct {
	/**
	 * Client ID
	 */
	ClientOid string `json:"clientOid"`
	/**
	 * Amount of currency placed
	 */
	Size string `json:"size"`
	/**
	 * 1: Kaiduo 2: Kaikong 3: Pingduo 4: Pingkong
	 */
	Side string `json:"side"`
	/**
	 * Order Type
	 */
	OrderType string `json:"orderType"`
	/**
	 * Entrusted price
	 */
	Price            string `json:"price"`
	TimeInForceValue string `json:"timeInForceValue"`
}

File: bitget-node-sdk-api/src/lib/model/mix/order/BatchOrdersReq.ts
import {PlaceOrderBaseParam} from './PlaceOrderBaseParam';

export class BatchOrdersReq {
    /**
     * Currency pair
     */
    private _symbol!:string;
    /**
     * Deposit currency
     */
    private _marginCoin!:string;
    /**
     * Order data list
     */
    private _orderDataList!:PlaceOrderBaseParam[]

    get symbol(): string {
        return this._symbol;
    }

    set symbol(value: string) {
        this._symbol = value;
    }

    get marginCoin(): string {
        return this._marginCoin;
    }

    set marginCoin(value: string) {
        this._marginCoin = value;
    }

    get orderDataList(): PlaceOrderBaseParam[] {
        return this._orderDataList;
    }

    set orderDataList(value: PlaceOrderBaseParam[]) {
        this._orderDataList = value;
    }
}